### PR TITLE
Add some checks for merged test groups

### PR DESCRIPTION
**Pull request by markples**
- CLRTest.Execute.Batch.targets
  - Fix CLRTestExecutionArguments (allows passing a filter argument to <mergedgroup>.cmd
  - Fix indenting in generated script
- XUnitWrapperGenerator
  - Move some generated code into (shared) ITestInfo.cs
  - Also display exception on stdout on failure
  - Add check that projects in merged test groups don't contain entry points
    - Can be a sign of code that is expected to run but won't be
    - Or can be a [Fact]/etc-less entry point in a ReqProcIso project but would then need to be fixed if ReqProcIso were removed
- Directory.Build.targets
  - Run XUnitWrapperGenerator on test projects in merged groups
  - Can't simply replace the conditional because we have XUnit-style tests outside of merged groups
- Update tests to conform to above checks plus a bit of opportunistic cleanup

### CODE REVIEW

File: src/tests/JIT/Regression/JitBlue/Runtime_75832/Runtime_75832.cs [hunk 1]
```diff
@@ -8,30 +8,11 @@
...
-        }
-        catch (Exception ex)
-        {
-            Console.WriteLine("FAIL: Caught {0}", ex.GetType().Name);
-        }
-        
-        return 101;
-    }
+    public static void TestEntryPoint()
+        => Assert.Throws<DivideByZeroException>(() => Test(0));
```
Comment (jakobbotsch) [inline src/tests/JIT/Regression/JitBlue/Runtime_75832/Runtime_75832.cs:12]: The general problem I have with these assert helpers is that they expand into a ton of unrelated code that I prefer not to see when I'm looking at the test. I stopped using these generally in JIT tests when I had to skip past tons of inlined `Assert.Equals` gunk while investigating my test.

It's probably ok for this test since most interesting stuff goes on inside `Test`.

File: src/tests/JIT/opt/Devirtualization/box2.cs [hunk 2]
```diff
@@ -17,7 +17,10 @@ static async Task<int> Main()
             // extra context argument.
             await new ValueTask<string>(Task.Delay(1).ContinueWith(_ => default(string))).ConfigureAwait(false);
         }
+    }
 
-        return 100;
+    public static void TestEntryPoint()
```
Comment (jakobbotsch) [inline src/tests/JIT/opt/Devirtualization/box2.cs:22]: `Fact`? Also, `Task.Run(TestTask)` should presumably be `Task.Run(TestTask).Wait()`.

Comment (markples) [reply src/tests/JIT/opt/Devirtualization/box2.cs]: thanks - fixed

File: src/tests/JIT/Regression/JitBlue/Runtime_75832/Runtime_75832.cs (hunk 1 above)
Comment (markples) [reply src/tests/JIT/Regression/JitBlue/Runtime_75832/Runtime_75832.cs:12]: Thanks for the feedback. This is interesting as I've had the opposite experience when looking at the source code (rather than debugging them) in that there is so much boilerplate in the way. That said, I don't plan on widespread changes to call Equals a bunch but just happened to notice how much boilerplate was needed for Throws while auditing recent test changes. I also came to the conclusion that Throws was "less" than Equals and friends.

It would be nice to have a helper that addresses both sides. I wonder if the xunit folks would accept NoInlining everywhere. Alternatively, we could build our own wrapper library, though it would be unfortunate to end up "not standard".

(Kind of related - [Theory] often ends up not saving much over a wrapper [Fact] method since you often end up needing to pass the expected output and explicitly check against it anyway. It would be nice to have the expected return in the [InlineData]/etc., but again it wouldn't be standard.

Comment (markples) [reply src/tests/JIT/Regression/JitBlue/Runtime_75832/Runtime_75832.cs:12]: It's also interesting how often tests define their own AssertEquals, etc.!

**Comment by markples** [comment]
@trylek @jkoritzinsky This starts to add some checks to avoid mistakes in our tests. PTAL - feedback especially welcome on how to do things better with Roslyn!
cc @dotnet/jit-contrib

upcoming - #89550, #89554, and possibly other things that we find to check

File: src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs [hunk 3]
```diff
@@ -112,40 +131,68 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
...
+        foreach (IMethodSymbol entryPoint in compData.PossibleEntryPoints)
+        {
+            context.ReportDiagnostic(Diagnostic.Create(
+                new DiagnosticDescriptor(
+                    "XUW1001",
+                    "No explicit entry point",
+                    "Projects in merged tests group should not contain entry points",
+                    "XUnitWrapperGenerator",
+                    DiagnosticSeverity.Warning,
+                    isEnabledByDefault: true),
```
Comment (jkoritzinsky) [inline src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs:191]: We should save this in a static readonly field (makes it easier to know which diagnostic ids are in use).

**Review by jkoritzinsky — APPROVED** [review]

File: src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs (hunk 3 above)
Comment (markples) [reply src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs]: Thanks. just fyi - I also noticed that I was generating SimpleRunners for all tests now and restricted the actual wrapper generation to ConsoleApplications (hopefully nothing is depending on libraries having entry points).  I think I will run outerloop+extraplatforms to verify.

**Comment by markples** [comment]
runtime is clean.  outerloop and extraplatforms seem to be #89545, #88808, and completely unrelated jobs

File: src/tests/Common/XUnitWrapperLibrary/TestSummary.cs [hunk 4]
```diff
@@ -133,8 +140,10 @@ public void WriteFooterToTempLog(StreamWriter tempLogSw)
         var result = new TestResult(name, containingTypeName, methodName, duration, null, null, output);
         _testResults.Add(result);
 
+        outTw.WriteLine($"{0:HH:mm:ss.fff} Passed test: {1}", System.DateTime.Now, name);
```
Comment (jakobbotsch) [inline src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:143]: @markples This line results in printing "HH:mm:ssfff Passed test: 1" always -- presumably it shouldn't be a string interpolation.

Comment (markples) [reply src/tests/Common/XUnitWrapperLibrary/TestSummary.cs:143]: Ah, that's leftover from the original
```
builder.AppendLine($"System.Console.WriteLine(\"{{0:HH:mm:ss.fff}} Running test: {{1}}\", System.DateTime.Now, {test.TestNameExpression});");
```
but was for the outer level. It's weird though; I thought that I looked at the output.
